src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -1199,6 +1202,17 @@ bool BTypeVisitor::VisitVarDecl(VarDecl *Decl) {
       }
       fe_.table_storage().Insert(global_path, table_it->second.dup());
       return true;
+    } else if(A->getName() == "maps/shared") {
+      if (table.name.substr(0, 2) == "__")
+        table.name = table.name.substr(2);
+      Path local_path({fe_.id(), table.name});
+      Path maps_ns_path({fe_.maps_ns(), table.name});
+      if (!fe_.table_storage().Find(local_path, table_it)) {
+        error(GET_BEGINLOC(Decl), "reference to undefined table");
+        return false;
+      }
+      fe_.table_storage().Insert(maps_ns_path, table_it->second.dup());
```

Comment:
This work for multiple maps, you can share any number of maps.

Table storage saves the map in a directory like structure, public tables are saved in the root, for each program there is a directory where privates tables are saved.

```
ebpf1:
BPF_PUBLIC(public_table)
BPF_TABLE(table1)

ebpf2:
BPF_TABLE(table1)
BPF_EXTERN(public_table) 
```

Would produce the following structure
```
/public_table
/ebpf1/public_table
/ebpf1/table1
/ebpf2/table1
/ebpf2/public_table
```

Shared tables are saved in a per namespace folder:

```
ebpf1 (namespace ns0):
BPF_SHARED(shared_table)
BPF_TABLE(table1)

ebpf2 (namespace ns0):
BPF_TABLE(table1)
BPF_EXTERN(shared_table) 

ebpf3 (namespace ns1):
BPF_SHARED(shared_table)
BPF_TABLE(table1)

ebpf4 (namespace ns1):
BPF_TABLE(table1)
BPF_EXTERN(shared_table) 
```

Would produce the following structure
```
/ns0/shared_table
/ns1/shared_table
/ebpf1/shared_table
/ebpf1/table1
/ebpf2/shared_table
/ebpf2/table1
/ebpf3/shared_table
/ebpf3/table1
/ebpf4/shared_table
/ebpf4/table1
```

In this case there are to different instances of shared_table, the first one in ns0 (ebpf1, ebpf2), the second one in ns1 (ebpf3, ebpf4).

Maybe I was too verbose, just wanted to let it very clear.

Reply:
Thanks for explanation. There is a near 0% chance that the namespace ID could be the same as ebpf ID in the above. I say near 0% since the ebpf ID is a random pointer value and name space in general is a human readable string. Do you think we should do further checking or just leave as is now?

Reply:
I don't think this is a real issue, but just in case I am going to add a prefix `ns` to all namespaces. It should be a 1 or 2 lines change.
```
...
/ns/ns0/shared_table
...
```